tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2129,3 +2132,126 @@ def test_upgrade_and_enable_basic_auth(self, base_release: tuple[int, int],
+            if offset["partition"] not in self.partition_offsets:
+                self.partition_offsets[offset["partition"]] = {}
+            self.partition_offsets[offset["partition"]][
+                offset["offset"]] = msg_ascii
+
+            return False
+
+        try:
+            wait_until(do_produce, timeout_sec=self.timeout_sec, backoff_sec=1)
+        except ducktape.errors.TimeoutError as ex:
+            self.done = True
+        except Exception as ex:
+            self.exception = ex
+
+
+class VerifiablePandaproxyConsumer(threading.Thread):
+    def __init__(self,
+                 topic: TopicSpec,
+                 consumer: Consumer,
+                 verifiable_producers: list[VerifiablePandaproxyProducer],
+                 timeout_sec: int,
+                 logger: TestLoggerMaker,
+                 auth_tuple: Optional[tuple[str, str]] = None):
+        threading.Thread.__init__(self)
+        self.daemon = True
+        self.topic = topic
+        self.consumer = consumer
+        self.verifiable_producers = verifiable_producers
+        self.timeout_sec = timeout_sec
+        self.logger = logger
+        self.auth_tuple = auth_tuple
+        self.result_raw = None
+        self.done = False
+        self.exception = None
+
+    def _filter_producer_by_partition_offset(self, partition: int,
+                                             offset: int):
+        try:
+            prod = next(
+                filter(
+                    lambda prod: partition in prod.partition_offsets and offset
+                    in prod.partition_offsets[partition],
+                    self.verifiable_producers))
+            return prod
+        except StopIteration:
+            raise RuntimeError(
+                f"Failed to find producer with partition-offset {partition}-{offset}"
+            )
+
+    def _fetch_message_by_partition_offset(self, partition: int, offset: int):
+        verifiable_producer = self._filter_producer_by_partition_offset(
+            partition, offset)
+        return verifiable_producer.partition_offsets[partition][offset]
+
+    def run(self):
+        sc_res = self.consumer.subscribe([self.topic.name],
+                                         auth=self.auth_tuple)
+        assert sc_res.status_code == requests.codes.no_content
+
+        def do_consume():
+            cf_res = self.consumer.fetch(
+                headers=HTTP_CONSUMER_FETCH_JSON_V2_HEADERS,
+                auth=self.auth_tuple)
+            assert cf_res.status_code == requests.codes.ok
+            fetch_result = cf_res.json()
+            self.logger.debug(f"fetch result {cf_res} {fetch_result}")
+
+            for r in fetch_result:
+                assert r["partition"] is not None
+                assert r["offset"] is not None
+                msg_ascii = self._fetch_message_by_partition_offset(
+                    r["partition"], r["offset"])
+                assert r["value"] == msg_ascii
```

Comment:
I recognize that this test does **not** use consumer offset commit or fetch. Instead the test runs consumer fetch. This is sufficient because `kafka::client::consumer_fetch` also uses `gated_retry_with_mitigation`. You can see in the redpanda logs that the fetch request is retried in the presence of network errors (RPC errors). So, the retry loop in consumer offset commit and fetch will also retry on network errors.